src/framework/components/anim/component.js

```diff
@@ -109,10 +106,27 @@ class AnimComponent extends Component {
         this.animationAssets = prevAnimationAssets;
         this.loadAnimationAssets();
         // assign the previous layer masks then rebind all anim targets
-        this.layers.forEach((layer, i) => layer.assignMask(prevMasks[i]));
+        this.layers.forEach((layer, i) => {
+            layer.mask = prevMasks[i];
+        });
         this.rebind();
     }
 
+    /**
```

Comment:
I wonder if the public API should be `getAnimationAsset(state)` that would return the asset id of the requested state? This would be more friendly to use, and would hide the implementation detail, allowing us to change it easier if needed.

Reply:
👍  This new function should live on the AnimComponentLayer then as you need both the layer name and state name to retrieve an asset.